documentation/blog/2025-06-24-mcp-jupyter-server/index.md

```diff
@@ -0,0 +1,158 @@
+---
+title: "MCP Jupyter: AI-Powered Machine Learning and Data Science"
+description: Enable AI agents to work directly with your Jupyter notebooks, leveraging persistent memory and stateful collaboration for enhanced ML and data science workflows
+date: 2025-06-24
+authors:
+  - damien
+  - dean
+  - harrison
+---
+
+![MCP Jupyter Server](data-goose.png)
+
+Machine learning and data science workflows are inherently iterative. You load data, explore patterns, build models, and refine your approach based on results. But traditional AI assistants lose context between interactions, forcing you to reload data and re-establish context repeatedly—making data-heavy development slow and expensive.
+
+The [**MCP Jupyter Server**](https://github.com/block/mcp-jupyter) solves this by enabling AI agents like Goose to work directly with your Jupyter notebooks, maintaining persistent memory and state across interactions while letting the AI interact with your data through code execution rather than raw data transfer.
+
+<!--truncate-->
+
+## The Memory and Context Problem
+
+Traditional AI coding assistants face a fundamental limitation: they lose context between code iterations. They also are not designed for adhoc work where the expectation is the code should run from start to finish. This creates several challenges for data-heavy ML workflows:
+
+- **Constant data reloading**: Each interaction requires re-loading or describing your dataset
+- **Expensive iterations**: Large datasets are slow and costly to process repeatedly
+- **Context size**: The raw data is also too large to be fit in context
+- **Not designed for adhoc work**: It is harder to have the model do some steps for you, and then let you take over and then later pass control back to the model
+
+This has made it difficult to use AI to help with data science workflows, which led to the creation of the MCP Jupyter Server.
+
+## A Better Approach: Persistent State
+
+The MCP Jupyter Server takes a different approach. Instead of losing context between interactions, it maintains persistent memory and lets the AI interact with your data through code execution. This creates a powerful paradigm:
```

Comment:
```suggestion
The MCP Jupyter Server takes a different approach. Instead of losing context between interactions, it maintains persistent memory and lets the AI interact with your data through Jupyter code cell execution. This creates a powerful paradigm:
```